### PR TITLE
Require agents to follow CONTRIBUTING.md git conventions in stack rules

### LINKED ISSUE #399: Require agents to follow CONTRIBUTING.md git conventions in stack rules
## Context

The four stack rule files in [rules/](https://github.com/awinogradov/code-assistants/tree/main/rules) are synced downstream as each consumer repository's `CLAUDE.md` — they are the instructions coding agents actually load. Today their only tie to [CONTRIBUTING.md](https://github.com/awinogradov/code-assistants/blob/main/CONTRIBUTING.md) is the fallback clause on the MANDATORY skill rules ("If the autopilot plugin is not installed, follow CONTRIBUTING.md") — nothing declares the CONTRIBUTING.md Git Workflow section as the binding standard for branch names, commit messages, PR titles/descriptions, and issue linking. In practice agents intermittently ignore those conventions when creating branches, commits, and PRs, and the violation only surfaces later when the contributing-check CI action or the review bot blocks the PR.

Related: #296 (closed), #317 (closed), #390 (closed)

## What

Add an explicit, mandatory Git Workflow rule block to all four `rules/*.md` files (`Bun.md`, `Bun+React+Tailwind.md`, `NodeJS+React.md`, `NodeJS+React+Tailwind.md`) that hard-requires following CONTRIBUTING.md for every branch, commit, PR, and issue operation — branch naming (`issue-<number>-<slug>` and special prefixes), Conventional Commits format and length limits, PR title/description rules, and magic-word issue linking — kept identical across the four files.

## Why

Catching a convention violation in CI wastes a full round-trip: the branch is already pushed, the PR is blocked, and the agent has to rename/reword and force-push. Encoding the restriction where agents read their instructions (the synced CLAUDE.md) prevents the violation at creation time instead of rejecting it at validation time, in this repo and every downstream consumer.

## Scope

- **In scope:**
  - A Git Workflow rule block added to all four `rules/*.md` files, byte-identical across them
  - Explicit references to the CONTRIBUTING.md sections that define each convention
- **Out of scope:**
  - Changes to CONTRIBUTING.md itself (it already defines the conventions)
  - Changes to autopilot skills (they already encode the conventions)
  - Changes to CI validation (contributing-check already enforces post hoc)

## Solution

Add a rule block to the git-workflow-related sections of each stack rules file stating that CONTRIBUTING.md is the binding standard for all git/GitHub write operations: read it before creating a branch, commit, PR, or issue, and follow its Branches, Commits, PR Title, PR Description, and Magic Words sections exactly. Keep the block identical across the four stack files (mirroring how the existing MANDATORY skill rules are duplicated), so the downstream agents-rules sync distributes it unchanged.

### PR DESCRIPTION
**Pull request by awinogradov**
Agents intermittently ignored [CONTRIBUTING.md](https://github.com/awinogradov/code-assistants/blob/main/CONTRIBUTING.md) conventions when creating branches, commits, and PRs whenever the autopilot skills were bypassed or unavailable. The stack rule files now bind those conventions directly in the instructions agents load as CLAUDE.md.

- Added a `## 15. Git Workflow` section to all four `rules/*.md` files, byte-identical, in the vacant slot between §14 Anti-Patterns and §16 AI Assistant Workflow — CONTRIBUTING.md is the binding standard, referenced per operation by section name with no restated rules, so CONTRIBUTING.md can evolve without the rule files going stale
- Bridged to §16.1: invoking the `Skill(autopilot:*)` commands satisfies the section
- Added a sync guard to [validate-plugins.ts](https://github.com/awinogradov/code-assistants/blob/main/scripts/validate-plugins.ts) that fails `bun run validate` when the section is missing from any rules file or differs between any two

---

**Issues:**

Closes #399

### CODE REVIEW

File: scripts/validate-plugins.ts [hunk 1]
```diff
@@ -100,23 +100,49 @@ async function validateTarget(t: Target): Promise<string | null> {
...
 }
 
+const rulesSectionHeading = "## 15. Git Workflow";
+
+/**
+ * The rules/<stack>.md files carry a duplicated Git Workflow section that ships
+ * downstream as each consumer's CLAUDE.md. Verify the section exists in every
+ * file and is byte-identical across them so the copies cannot drift.
+ */
+async function validateRulesSectionSync(): Promise<string | null> {
```
[the remark this answers is not in the document]
Comment (awinogradov) [reply scripts/validate-plugins.ts:110]: Keeping both as-is — rationale:

**Scope** — the out-of-scope item in #399 was changes to CI validation of the conventions themselves (the [contributing-check](https://github.com/awinogradov/code-assistants/blob/main/.github/actions/contributing-check/README.md) action's post-hoc enforcement). The issue's in-scope **What** requires the section to be *kept identical across the four files*, and `validateRulesSectionSync` is what makes that requirement durable instead of a one-time check — the same pattern that guards the duplicated [RFC-0001](https://github.com/awinogradov/code-assistants/blob/main/rfc/0001-reference-formatting.md) block via `referenceFormattingSync`.

**Test coverage** — `scripts/` is not a workspace member, so a `*.test.ts` beside [validate-plugins.ts](https://github.com/awinogradov/code-assistants/blob/main/scripts/validate-plugins.ts) would never execute under `turbo run test` in CI; a test file CI never runs is worse than none. The guard itself executes on every PR via the Validate check (`bun run validate`) and on every markdown commit via lint-staged, and the edge cases were exercised on this branch: a one-word divergence in one file fails with the offending path and exit 1, and a `--files` invocation with a non-rules path skips the check. Real unit coverage would mean promoting `scripts/` to a workspace member with its own test setup — a follow-up issue if wanted, not this PR.

[the remark this answers is not in the document]
Comment (awinogradov) [reply scripts/validate-plugins.ts:110]: Half-conceded, with one correction:

**lint-staged does exist** — the config is [lint-staged.config.ts](https://github.com/awinogradov/code-assistants/blob/main/lint-staged.config.ts) at the repo root (a supported config form; there is no `package.json` block or `.lintstagedrc` to find). Its `**/*.md` entry runs `bun scripts/validate-plugins.ts --files` on every markdown commit via the husky pre-commit hook — both commits on this branch show the guard executing in the hook output.

**Conceded on CI wiring** — you're right that no PR workflow runs `bun scripts/validate-plugins.ts`; my earlier claim that it runs via a CI check was wrong. That's a pre-existing gap for the whole script (plugin frontmatter/manifest validation isn't CI-run either), not something introduced by this guard, so wiring `bun run validate` into a workflow is a repo-wide follow-up rather than this PR. Until then the guard's enforcement tier is the pre-commit hook — the same tier that already protects the plugin manifests.
